### PR TITLE
[CBRD-23842] Fixed bugs when delete operation on REC_HOME are not extracted

### PR DESCRIPTION
**Pull request by joohok**
http://jira.cubrid.org/browse/CBRD-23842

Describe here

### CODE REVIEW

File: src/transaction/log_impl.h [hunk 1]
```diff
@@ -824,13 +824,13 @@ typedef enum cdc_producer_state
   CDC_PRODUCER_STATE_DEAD
 } CDC_PRODUCER_STATE;
 
-typedef enum cdc_producer_request
+typedef enum cdc_request
 {
-  CDC_REQUEST_PRODUCER_IS_DEAD,
-  CDC_REQUEST_PRODUCER_IS_WAITED,
+  CDC_REQUEST_PRODUCER_TO_WAIT,
+  CDC_REQUEST_PRODUCER_TO_BE_DEAD,
```
Comment (joohok) [inline src/transaction/log_impl.h:837]: separate producer, consumer request

File: src/transaction/log_manager.c [hunk 2]
```diff
@@ -10876,17 +10876,20 @@ cdc_loginfo_producer_execute (cubthread::entry & thread_ref)
 
   int error = NO_ERROR;
 
-  while (cdc_Gl.producer.state != CDC_PRODUCER_STATE_DEAD)
+  while (cdc_Gl.producer.request != CDC_REQUEST_PRODUCER_TO_BE_DEAD)
```
Comment (joohok) [inline src/transaction/log_manager.c:10881]: request, state initialization

File: src/transaction/log_manager.c [hunk 3]
```diff
@@ -10876,17 +10876,20 @@ cdc_loginfo_producer_execute (cubthread::entry & thread_ref)
...
+
+	  cdc_Gl.producer.state = CDC_PRODUCER_STATE_WAIT;
+	  cdc_Gl.producer.request = CDC_REQUEST_NONE;
 
 	  pthread_mutex_lock (&cdc_Gl.producer.lock);
 	  pthread_cond_wait (&cdc_Gl.producer.wait_cond, &cdc_Gl.producer.lock);
 	  pthread_mutex_unlock (&cdc_Gl.producer.lock);
 
+	  cdc_Gl.producer.state = CDC_PRODUCER_STATE_RUN;
 	  cdc_wakeup_consumer ();
```
Comment (joohok) [inline src/transaction/log_manager.c]: remove this wake up consumer process

File: src/transaction/log_manager.c [hunk 4]
```diff
@@ -10898,10 +10901,15 @@ cdc_loginfo_producer_execute (cubthread::entry & thread_ref)
 
 	  cdc_pause_consumer ();
 
+	  cdc_Gl.producer.state = CDC_PRODUCER_STATE_WAIT;
+	  cdc_Gl.producer.request = CDC_REQUEST_NONE;
```
Comment (joohok) [inline src/transaction/log_manager.c]: remove request init

File: src/transaction/log_manager.c [hunk 5]
```diff
@@ -10898,10 +10901,15 @@ cdc_loginfo_producer_execute (cubthread::entry & thread_ref)
 
 	  cdc_pause_consumer ();
 
+	  cdc_Gl.producer.state = CDC_PRODUCER_STATE_WAIT;
```
Comment (joohok) [inline src/transaction/log_manager.c:10903]: move before cdc_pause_consumer ();

File: src/transaction/log_manager.c [hunk 6]
```diff
@@ -13973,14 +13984,13 @@ cdc_make_loginfo (THREAD_ENTRY * thread_p, LOG_LSA * start_lsa)
...
   LSA_COPY (&cdc_Gl.consumer.next_lsa, start_lsa);	/* stores next lsa to consume */
 
-  while (cdc_Gl.consumer.request == CDC_REQUEST_CONSUMER_TO_WAIT)
+  if (cdc_Gl.consumer.request == CDC_REQUEST_CONSUMER_TO_WAIT)
     {
-
       cdc_wakeup_producer ();
 
-      if (cdc_Gl.consumer.request == CDC_REQUEST_NONE || cdc_Gl.consumer.consumed_queue_size == 0)
+      while (cdc_Gl.consumer.request != CDC_REQUEST_NONE)
```
Comment (joohok) [inline src/transaction/log_manager.c]: CDC_REQUEST_CONSUMER_RUN

Comment (joohok) [reply src/transaction/log_manager.c]: cdc_Gl.consumer.request : NONE -> RUN -> NONE

**Review by hornetmj — APPROVED** [review]

**Pull request merged**